discovery/kubernetes/node.go

```diff
@@ -207,9 +209,15 @@ func nodeAddress(node *apiv1.Node) (string, map[apiv1.NodeAddressType][]string,
 		m[a.Type] = append(m[a.Type], a.Address)
 	}
 
+	if addresses, ok := m[apiv1.NodeInternalDNS]; ok {
```

Comment:
I think we should prefer IP to DNS, as that avoids having to do additional DNS lookups.

Reply:
Oh yes, that makes sense. Will do.